modules/spatiotemporal/test/test_make_graph.py

```diff
@@ -0,0 +1,170 @@
+import IMP
+import IMP.test
+import IMP.spatiotemporal as spatiotemporal
+import IMP.spatiotemporal.graphNode as graphNode
+import shutil
+import os
+import itertools
+import numpy as np
+
+def setup_system():
+    # Input variables.
+    dict = {'0min': 2, '5min': 3, '10min': 2}
+    subcomplexes = ['A1', 'A2', 'B1', 'B2']
+    # exp_comp_map is a dictionary that describes protein stoicheometery. The key describes the protein, which should correspond to names within the expected_subcomplexes. For each of these proteins, a csv file should be provided with protein copy number data
+    exp_comp = {'A': 'exp_comp_A.csv', 'B': 'exp_comp_B.csv'}
+    return dict, subcomplexes, exp_comp
+
+class Tests(IMP.test.TestCase):
+
+    def test_graph_setup(self):
+        # test setting up a graph. Tests functionality of graphNode.py
+        state_dict, expected_subcomplexes, exp_comp_map = setup_system()
+                # set input dir
+        with IMP.test.temporary_directory() as tmpdir:
+            input_dir = os.path.join(tmpdir, 'data')
+        shutil.copytree(self.get_input_file_name('data'), input_dir)
```

Comment:
I think your whole function needs to be inside the `with` block. When that block ends, `tmpdir` (and thus `input_dir`) will be deleted.